README.md

```diff
@@ -1,3 +1,5 @@
+![build status](https://github.com/atomic14/diy-esp32-epub-reader/actions/workflows/build-test-on-push.yml/badge.svg)
+
```

Comment:
Nice one. Never did this, what is the way to run the CI test?

Reply:
locally you can run:

```
pio test -e native
```

With the GitHub action that tests will run whenever we push to the repository - you can see it on this PR - there's a section "All checks have passed" if you expand that out you can see the workflow that was run - `.github/workflows/build-test-on-push.yml`

The tests are in the `tests` folder - I've just got some very basic ones at the moment, but we can add more. The only tricky bit is making sure that the code you are trying to tests stays "clean" of any hardware-specific code.

There is a way to run tests on the device - but obviously, that doesn't work in a CI environment.

Reply:
Nice:

Testing...
test/tests.cpp:13:test_xml_parser       [PASSED]
Failed to read item test.png
Failed to read item HTML/test.png
test/tests.cpp:14:test_parser   [PASSED]
Missing cover
test/tests.cpp:15:test_epub_no_oebps_load       [PASSED]
test/tests.cpp:16:test_epub_load        [PASSED]
test/tests.cpp:17:test_epub_relative_image_paths        [PASSED]
test/tests.cpp:18:test_html_entity_replacement  [PASSED]

-----------------------
6 Tests 0 Failures 0 Ignored